pkg/dns/treecache.go

```diff
@@ -59,61 +50,33 @@ func (cache *TreeCache) Serialize() (string, error) {
 }
 
 func (cache *TreeCache) setEntry(key string, val interface{}, path ...string) {
```

Comment:
I don't think the comment would make sense here. the treecache is not thread-safe.
It could be used by the same thread/go-routine without locking,
or by multiple threads, using a simple mutex or rwmutex. 
it is up to the caller to choose..